lib/rules/function-url-quotes/index.mjs

```diff
@@ -155,35 +168,53 @@ const rule = (primary, secondaryOptions) => {
 		/**
 		 * @param {string} args
 		 * @param {number} index
-		 * @returns {Pick<Problem, 'index' | 'endIndex'> & { messageType: keyof messages } | undefined}
+		 * @returns {Pick<Problem, 'index' | 'endIndex'> & { messageType: keyof messages, modifierIndex?: number } | undefined}
 		 */
 		function checkArgs(args, index) {
-			const leftTrimmedArgs = args.trimStart();
+			const trimmedArgs = args.trim();
 
-			if (!isStandardSyntaxUrl(leftTrimmedArgs)) return;
+			if (!isStandardSyntaxUrl(trimmedArgs)) return;
 
 			let expectQuotes = primary === 'always';
 
-			if (exceptEmpty && emptyArgumentPatterns.has(args.trim())) {
+			if (exceptEmpty && emptyArgumentPatterns.has(trimmedArgs)) {
 				expectQuotes = !expectQuotes;
 			}
 
-			const hasQuotes = leftTrimmedArgs.startsWith("'") || leftTrimmedArgs.startsWith('"');
+			const hasQuotes = trimmedArgs.startsWith("'") || trimmedArgs.startsWith('"');
 
 			if (expectQuotes && hasQuotes) return;
 
 			if (!expectQuotes && !hasQuotes) return;
 
+			const modifierIndex = findModifierIndex(trimmedArgs, hasQuotes);
+
+			if (hasQuotes && modifierIndex) return;
+
 			const messageType = expectQuotes ? 'expected' : 'rejected';
-			const reportIndex = index + args.length - leftTrimmedArgs.length;
-			const reportEndIndex = index + args.length;
+			const reportIndex = index + args.length - args.trimStart().length;
+			const reportEndIndex = index + (modifierIndex ? modifierIndex - 1 : args.length);
 
 			return {
 				messageType,
 				index: reportIndex,
 				endIndex: reportEndIndex,
+				modifierIndex,
 			};
 		}
+
+		/**
+		 * @param {string} args
+		 * @param {boolean} startsWithQuote
+		 * @returns {number | undefined}
+		 */
+		function findModifierIndex(args, startsWithQuote) {
```

Comment:
note

We currently don't have test cases that use `var()` hence I suppose we don't support the indirection. You don't have to change anything; we should document that in the README though.
i.e. `v` !== `'` nor `"`